pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -207,11 +206,7 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 			if val, ok := pod.pod.Annotations[kueuealpha.PodGroupPodIndexLabelAnnotation]; ok {
 				pod.pod.Labels[kueuealpha.PodGroupPodIndexLabel] = pod.pod.Labels[val]
 			}
-
-			if jobframework.PodSetTopologyRequest(&pod.pod.ObjectMeta, ptr.To(kueuealpha.PodGroupPodIndexLabel), nil, nil) != nil {
-				pod.pod.Labels[kueuealpha.TASLabel] = "true"
-				utilpod.Gate(&pod.pod, kueuealpha.TopologySchedulingGate)
-			}
+			utilpod.Gate(&pod.pod, kueuealpha.TopologySchedulingGate)
```

Comment:
Isn't this increasing the dequeue time for non-TAS workloads?

Reply:
Yeah, I see where you are coming from, but I think this is the only way to delay the decision if TAS-scheduling is used or non-TAS later - because in k8s we cannot add scheduling gates later.

However, I think this is perfectly ok, because:
1. this only affects the Pod integration, and pod-based integrations.
2. it does not increase the time to schedule non-TAS pods, as the gate is removed together with the main "admission" gate, see [code](https://github.com/kubernetes-sigs/kueue/pull/4591/files#diff-dfb49586a8522fa91d733051fd3b7e4b3ff174907898cf249ab16e2620976a5dR1346-R1351), and the experiment 3 with a watch in the [comment](https://github.com/kubernetes-sigs/kueue/pull/4591#issuecomment-2724938368)

Reply:
> because in k8s we cannot add scheduling gates later.

Yeah, that's right.
Actually, I was thinking of an alternative solution last weekend, a bit. But there is no way...
This looks like an egg and chicken problem...

At least, shouldn't we guard this behavior by FeatureGate?

Reply:
> At least, shouldn't we guard this behavior by FeatureGate?

For sure, it remains guarded: https://github.com/kubernetes-sigs/kueue/pull/4591/files#diff-8653db7737ac7710352223a2af7723aa2def7cc433c73b1d151c7502d2a9af17R205-L214

Reply:
Oh, you're right.
Additionally, I want to add TAS scenario to scheduling perf for TAS beta graduation.
As we know, TAS is greedy consume memory, CPU and scheduling time.

Could you open KEP update PR?

Reply:
Sure, opened the issue: https://github.com/kubernetes-sigs/kueue/issues/4634.

I will follow up with KEP update soon.